providers/google/src/airflow/providers/google/cloud/operators/ray.py

```diff
@@ -0,0 +1,418 @@
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.cluster_address = cluster_address
+        self.job_id = job_id
+
+    def execute(self, context: Context):
+        try:
+            self.log.info("Deleting Job %s on a Ray cluster...", self.job_id)
+            is_deleted = self.hook.delete_job(
+                cluster_address=self.cluster_address,
+                job_id=self.job_id,
+            )
+            if is_deleted:
+                self.log.info("Job was successfully deleted.")
+                return
+            raise AirflowException("Some error happened during deleting the Job. Exiting.")
+        except NotFound:
+            raise AirflowException("Job with specified id was not found on cluster.")
+
+
+class RayGetJobInfoOperator(RayJobBaseOperator):
+    """
+    Get the latest status and other information associated with a Job on Ray cluster.
+
+    :param cluster_address: Required. Either (1) the address of the Ray cluster, or (2) the HTTP address
+        of the dashboard server on the head node, e.g. "http://<head-node-ip>:8265".
+        In case (1) it must be specified as an address that can be passed to
+        ray.init(), e.g. a Ray Client address (ray://<head_node_host>:10001),
+        or "auto", or "localhost:<port>".
+    :param job_id: Required. The job ID or submission ID for the job to be stopped.
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    """
+
+    template_fields: Sequence[str] = tuple(
+        {"cluster_address", "job_id"} | set(RayJobBaseOperator.template_fields)
+    )
+    operator_extra_links = ()
+
+    def __init__(
+        self,
+        cluster_address: str,
+        job_id: str,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.cluster_address = cluster_address
+        self.job_id = job_id
+
+    def execute(self, context: Context):
+        try:
+            self.log.info("Retrieving information about Job %s on a Ray cluster...", self.job_id)
+            job_info = self.hook.get_job_info(
+                cluster_address=self.cluster_address,
+                job_id=self.job_id,
+            )
+            self.log.info("Job information:\n %s \n", job_info)
+            ray_job_dict = self.hook.serialize_job_obj(job_info)
```

Comment:
Could be outside of try except block and covered by separate try except as NotFound exception not covers possible serialization problems